ducktape/cluster/remoteaccount.py

```diff
@@ -250,6 +258,7 @@ def _can_ping_url(self, url, headers):
         except Exception:
             return False
 
+    @retry(retry_on_exception=retry_if_ssh_exception, wait_exponential_multiplier=1000, wait_exponential_max=60000)
```

Comment:
I'm a bit concerned about adding un-adjustable retries to the framework itself permanently - I mean we already know that ssh is flaky, but don't really know why - so while retries will possibly make the tests pass, they won't address the root of the problem. Especially them not being is what doesn't exactly *feel* right.

Also, this doesn't address ssh_capture and ssh_output methods. 

Also I wonder if we could adjust a connection timeout on `self.ssh_client.exec_command` and `self.ssh_client.open_session` and the like instead of retries.

Reply:
> I'm a bit concerned about adding un-adjustable retries to the framework itself permanently - I mean we already know that ssh is flaky, but don't really know why - so while retries will possibly make the tests pass, they won't address the root of the problem.

You're hard pressed to convince me that adding retries here *isn't* a good idea when testing distributed systems. If this does end up addressing the issues, that means that we simply needed exponential backoff when calling commands over SSH - if it doesn't, well it makes the framework more resilient anyways.

> Especially them not being is what doesn't exactly feel right.

I don't understand what you're saying here.

> Also, this doesn't address ssh_capture and ssh_output methods.

Sure, I'll add these.

> Also I wonder if we could adjust a connection timeout on self.ssh_client.exec_command and self.ssh_client.open_session and the like instead of retries.

I acknowledge that this is me taking shots in the dark to try and address a problem we don't fully understand, but we should be methodical with adjusting changes here. Retrying IMO is more clean than increasing TCP timeouts, and we should try that before we increase timeouts.

Reply:
>Especially them not being is what doesn't exactly feel right.
Especially them not being *adjustable* is what doesn't exactly feel right. :)

Reply:
I think it's fine to add some retries, definitely up for debate though is what the max timeouts should be. Tests can already take quite a long time, adding retries only extends that, and depending on how fast the ssh failures are could make them take a *lot* longer.

I think this boils down to the tradeoff between test cost in the face of network failures vs developer time. For a change like this, it'd probably be a good idea to get an idea of what execution time changes look like over a statistically significant number of runs with the change.

Reply:
>adding retries here isn't a good idea when testing distributed systems
Yeah, nothing against retries in general, just not sure adding unconditional and unconfigurable retries on the lowest-level api method in the framework is a good idea. Consider the (hypothetical) case where a test would expect ssh connection to fail. In most cases you probably want some retries, but you should be able to adjust them or turn them off.
>If this does end up addressing the issues, that means that we simply needed exponential backoff when calling commands over SSH - if it doesn't, well it makes the framework more resilient anyways.
I would argue that we need to understand the nature and the cause of our issues with ssh before applying solutions. If it's actually flaky then retries would help, if the issue is of a different kind, then this will simply make the test run longer and fail anyways. 
Again, nothing against retries in general, but we need to understand the problem before applying the solution, all I'm saying :)